### ISSUE TITLE
Codegenerator produces errors in generics

### ISSUE THREAD

**Issue by wiezykrwi** [dotnet/orleans#5877]
I am using Microsoft.Orleans.CodeGenerator.MSBuild to generate proxies in a netstandard2.0 "contracts" library (shared library among clients to connect to orleans).
I recently updated to the latest stable version 2.4.1 from 2.3.6.

Where it used to generate 
`global::System.Collections.Generic.List<global::Our.Project.Namespace.SomeObject>`
It now generates
`global::System.Collections.Generic.List<SomeObject>`

Which doesn't compile.
Reverting back to 2.3.6 fixes the issue.

**Comment by ReubenBond** [comment]
@wiezykrwi do you have a project to reproduce the issue?

**Comment by wiezykrwi** [comment]
Ok so I created this very simple demo project: https://github.com/wiezykrwi/Orleans5877
Even stranger is that the issue also pops up in v2.3.6 (which seems logical since you mentioned on gitter that there were no notable changes in the code responsible for generation).

I was successfully and consistently able to reproduce the issue by following these steps:
1) change the version of the Microsoft.Orleans.CodeGenerator.MSBuild package (I alternate between 2.3.6 and 2.4.1)
2) close visual studio and delete all bin and obj folder (I use the included clean.ps1 script)
3) restart visual studio and build

You should get an error like this:
The type or namespace name 'Test' could not be found (are you missing a using directive or an assembly reference?)
@ReubenBond let me know if there is anything I can do to help!

**Comment by ReubenBond** [comment]
Apologies for the long delay, @wiezykrwi. I'm trying to reproduce this, following the steps you laid out. I'm not able to reproduce it yet. Is there anything else I should try? Which version of VS are you using? I'm using the latest 2019

**Comment by wiezykrwi** [comment]
Huh that's very strange @ReubenBond, I was able to reproduce the issue very consistently (and we recently had a round of the same problem in the office, and I was able to resolve it using the approach outlined above).

I use Visual Studio 2019 Preview (Enterprise), but most people in the office use regular Visual Studio 2019 (Enterprise) and they all had the same issues.

I just got a new laptop, I'll pick this up on the new machine on a stable and fresh Visual Studio 2019 (Enterprise) installation, asap.

**Comment by ReubenBond** [comment]
@wiezykrwi were you able to reproduce it?

**Comment by wiezykrwi** [comment]
Hey @ReubenBond sorry for the long delay - I've been otherwise preoccupied.
I just picked this up on a clean laptop, only installed Visual Studio Enterprise 2019 (with resharper) and I can't seem to get rid of the issue.
I don't do a lot more than is described in my previous post of Aug 22nd, I just cloned the project, first build worked, downgraded the version and now anything I do comes down to me not being able to build the project any more.
